syntax = "proto3";

package tvsc.comms.tdma.proto;

message TimeSlot {
  enum Role {
    GUARD_INTERVAL = 0;
    NODE_TX = 1;
    ASSOCIATION = 2;
    BLACKOUT = 3;
  }

  uint32 duration_us = 1;
  Role role = 2;
  uint64 slot_owner_id = 3;
}

message Frame {
  uint64 time_announced_us = 1;
  uint64 frame_start_time_us = 2;
  repeated TimeSlot time_slots = 3;
  uint64 base_station_id = 4;
}
